src/atomate2/common/jobs/elastic.py

```diff
@@ -200,21 +201,42 @@ def fit_elastic_tensor(
     allow_elastically_unstable_structs : bool
         Whether to allow the ElasticDocument to still complete in the event that
         the structure is elastically unstable.
+    max_failed_deformations: int or float
+        Maximum number of deformations allowed to fail to proceed with the fitting
+        of the elastic tensor. If an int the absolute number of deformations. If
+        a float between 0 an 1 the maximum fraction of deformations. If None any
+        number of deformations allowed.
     """
     stresses = []
     deformations = []
     uuids = []
     job_dirs = []
+    failed_uuids = []
     for data in deformation_data:
         # stress could be none if the deformation calculation failed
         if data["stress"] is None:
+            failed_uuids.append(data["uuid"])
             continue
 
         stresses.append(Stress(data["stress"]))
         deformations.append(Deformation(data["deformation"]))
         uuids.append(data["uuid"])
         job_dirs.append(data["job_dir"])
 
+    if max_failed_deformations is not None:
+        if 0 < max_failed_deformations < 1:
+            fraction_failed = len(failed_uuids) / len(deformation_data)
+            if fraction_failed > max_failed_deformations:
```

Comment:
This is a comparison of a fraction with something that looks like an int, right?